packages/lib-classifier/src/store/TutorialStore.spec.js

```diff
@@ -334,4 +374,108 @@ describe('Model > TutorialStore', function () {
       }).then(done, done)
     })
   })
+
+  describe('Actions > resetSeen', function () {
+    it('should reset only the tutorial seen time if tutorial is the kind', function () {
+      const seen = new Date().toISOString()
+      rootStore = RootStore.create({
+        tutorials: TutorialStore.create({ tutorialSeenTime: seen, miniCourseSeenTime: seen }),
+        workflows: WorkflowStore.create()
+      }, {
+        authClient: authClientStubWithoutUser, client: clientStub()
+      })
+
+      rootStore.tutorials.resetSeen('tutorial')
+      expect(rootStore.tutorials.tutorialSeenTime).to.be.undefined
+      expect(rootStore.tutorials.miniCourseSeenTime).to.equal(seen)
+    })
+
+    it('should reset only the tutorial seen time if mini-course is the kind', function () {
+      const seen = new Date().toISOString()
+      rootStore = RootStore.create({
+        tutorials: TutorialStore.create({ tutorialSeenTime: seen, miniCourseSeenTime: seen }),
+        workflows: WorkflowStore.create()
+      }, {
+        authClient: authClientStubWithoutUser, client: clientStub()
+      })
+
+      rootStore.tutorials.resetSeen('mini-course')
+      expect(rootStore.tutorials.tutorialSeenTime).to.equal(seen)
+      expect(rootStore.tutorials.miniCourseSeenTime).to.be.undefined
+    })
+
+    it('should reset both seen times if there is no defined kind', function () {
+      const seen = new Date().toISOString()
+      rootStore = RootStore.create({
+        tutorials: TutorialStore.create({ tutorialSeenTime: seen, miniCourseSeenTime: seen }),
+        workflows: WorkflowStore.create()
+      }, {
+        authClient: authClientStubWithoutUser, client: clientStub()
+      })
+
+      rootStore.tutorials.resetSeen()
+      expect(rootStore.tutorials.tutorialSeenTime).to.be.undefined
+      expect(rootStore.tutorials.miniCourseSeenTime).to.be.undefined
+    })
+  })
+
+  describe('Actions > setSeenTime', function () {
```

Comment:
The tests for `setActiveTutorial()` could maybe be rolled into this block, since we expect that to update the seen time (if I'm following this correctly.)

Reply:
This block is testing the behavior of `setSeenTime` in isolation. The test in the block for `setActiveTutorial` just has one expectation checking to see `setSeenTime` is called, not what it's doing.

Reply:
I understand that, but I don't think you should be testing implementation details, only expected state. Otherwise, you have to rewrite tests when implementation changes, even if the outcome is the same. I think that makes it harder to refactor since you can't test if behaviour has changed.

Reply:
I think there's some reason to test the details, mostly to check an action is only called once and with the correct parameters as expected. Only testing the expected state may not catch an action calling another multiple times. Is there a better way to test those kinds of expectations?

Reply:
Good point. If it's a public method, so might reasonably called by a component, then I think it's fair to spec its behaviour. If private, then I'd keep it hidden from tests.

Reply:
`setActiveTutorial` is used by the tabs in the task area when clicking on the tutorial tab, so it's public. `setSeenTime` is not.

Reply:
It'd be useful if there was a way we could distinguish those kinds of actions. Maybe through naming convention?

Reply:
If `setSeenTime` isn't meant to be directly accessed, then I think it can be kept private by removing it from the returned actions object. In the collections store, I've defined private functions for fetching/creating collections but I don't return them and I only test them indirectly via public actions.